components/ArticlePage.jsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { getArticleById } from "../api";
+
+function ArticlePage() {
+  const [article, setArticle] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const navigate = useNavigate();
+  const { articleid } = useParams();
+
+  useEffect(() => {
+    const loadArticle = async () => {
+      try {
+        const articleData = await getArticleById(articleid);
+        setArticle(articleData);
+        setLoading(false);
+      } catch (error) {
+        console.error("Error fetching article:", error);
```

Comment:
You could potentially have an error state here which updates to true, then amend your code on line 29 if you like 🙂 
We should also be setting Loading to be false in this block too 👍